river/Output.zig

```diff
@@ -185,227 +163,259 @@ pub fn sendViewTags(self: Self) void {
+/// Execute an external layout function, parse its output and apply the layout
+/// to the output.
+fn layoutExternal(self: *Self, visible_count: u32, output_tags: u32) bool {
     const border_width = self.root.server.config.border_width;
     const view_padding = self.root.server.config.view_padding;
     const outer_padding = self.root.server.config.outer_padding;
+    const xy_offset = @intCast(i32, border_width + outer_padding + view_padding);
+    const delta_size = (border_width + view_padding) * 2;
+    const layout_width = @intCast(u32, self.usable_box.width) - outer_padding * 2;
+    const layout_height = @intCast(u32, self.usable_box.height) - outer_padding * 2;
 
-    const layout_width = @intCast(u32, self.usable_box.width) -
-        (outer_padding * 2) - (border_width * 2) - (view_padding * 2);
-    const layout_height = @intCast(u32, self.usable_box.height) -
-        (outer_padding * 2) - (border_width * 2) - (view_padding * 2);
-    const x_offset = self.usable_box.x + @intCast(i32, outer_padding + border_width + view_padding);
-    const y_offset = self.usable_box.y + @intCast(i32, outer_padding + border_width + view_padding);
+    const visible_count_str = std.fmt.allocPrint(self.root.server.allocator, "{}", .{visible_count}) catch |err| switch (err) {
+        error.OutOfMemory => {
+            @panic("out of memory");
+        },
+    };
+    const master_count_str = std.fmt.allocPrint(self.root.server.allocator, "{}", .{self.master_count}) catch |err| switch (err) {
+        error.OutOfMemory => {
+            @panic("out of memory");
+        },
+    };
+    const master_factor_str = std.fmt.allocPrint(self.root.server.allocator, "{d}", .{self.master_factor}) catch |err| switch (err) {
+        error.OutOfMemory => {
+            @panic("out of memory");
+        },
+    };
+    const layout_width_str = std.fmt.allocPrint(self.root.server.allocator, "{}", .{layout_width}) catch |err| switch (err) {
+        error.OutOfMemory => {
+            @panic("out of memory");
+        },
+    };
+    const layout_height_str = std.fmt.allocPrint(self.root.server.allocator, "{}", .{layout_height}) catch |err| switch (err) {
+        error.OutOfMemory => {
+            @panic("out of memory");
+        },
+    };
 
-    var i: u32 = 0;
-    var it = ViewStack(View).pendingIterator(self.views.first, output_tags);
-    while (it.next()) |node| {
-        const view = &node.view;
+    // Create command to be executed. All commands are executed via sh and the
+    // five layout parameters are appended to the end.
+    const layout_command = std.mem.join(self.root.server.allocator, " ", &[_][]const u8{
+        self.layout,
+        visible_count_str,
+        master_count_str,
+        master_factor_str,
+        layout_width_str,
+        layout_height_str,
+    }) catch |err| {
+        Log.Error.log("Error assemblind layout command.", .{});
+        return false;
+    };
```

Comment:
add a 
```zig
defer allocator.free(layout_command);
```
here and remove all the calls farther down.

Also I think an 
```zig
const allocator = self.root.server.allocator;
```
at the top of the function would be justified.